pkg/drivers/kic/oci/network.go

```diff
@@ -168,42 +149,152 @@ func dockerContainerIP(name string) (string, string, error) {
 	return ips[0], ips[1], nil
 }
 
-// CreateNetwork creates a network
-func CreateNetwork(name, ipRange, gateway string) error {
+// CreateNetwork creates a network returns subnet and error
+func CreateNetwork(name string) (*net.IPNet, error) {
 	// check if the network already exists
-	if networkExists(name) {
-		return nil
+	subnet, _, err := dockerNetworkInspect(name)
+	if err == nil {
+		return subnet, nil
 	}
 
-	subnet := fmt.Sprintf("--subnet=%s", ipRange)
-	_, err := runCmd(exec.Command(Docker, "network", "create", "--driver=bridge", subnet, "--gateway", gateway, name))
+	// simple way to create networks, subnet is taken, try one bigger
+	attempt := 0
+	_, subnet, err = net.ParseCIDR(defaultSubnet)
 	if err != nil {
-		return errors.Wrapf(err, "error creating network")
+		return nil, errors.Wrapf(err, "parse default subnet %s", defaultSubnet)
+	}
+
+	err = attemptCreateNework(subnet, name)
+	if err != nil {
+		if err == ErrNetworkSubnetTaken {
+			// try up tp 13 times
+			for attempt < 13 {
+				attempt++
+				glog.Infof("Couldn't create network %q at %q subnet will try again a new subnet ...", name, subnet)
+				// increase 2nd digit by 10 each time
+				subnet.IP.To4()[2] += 10
+				err := attemptCreateNework(subnet, name)
+				if err == nil {
+					return subnet, nil
+				}
+				if err == ErrNetworkSubnetTaken {
+					continue
+				}
+			}
+		}
+		return nil, errors.Wrapf(err, "error creating network")
+
 	}
+	return subnet, nil
+}
 
+func attemptCreateNework(subnet *net.IPNet, name string) error {
+	gateway := subnet.IP.To4()
+	gateway[3]++ // first ip for gateway
+	glog.Infof("attempt to create network %q with subnet: %s and gateway %s...", subnet, name, gateway)
+	rr, err := runCmd(exec.Command(Docker, "network", "create", "--driver=bridge", fmt.Sprintf("--subnet=%s", subnet), fmt.Sprintf("--gateway=%s", gateway), "-o", "com.docker.network.bridge.enable_ip_masquerade=true", fmt.Sprintf("--label=%s=%s", CreatedByLabelKey, "true"), name))
+	if err != nil {
+		if strings.Contains(rr.Output(), "Pool overlaps with other one on this address space") {
```

Comment:
Ouch. Is it possible that Docker has exit codes we can use instead? I guarantee this will break some day when someone fixes the bad grammar usage in this error message.

Reply:
the exit code is 1 and it doesnt seem to be helpful
```
med@xmac:~/workspace/minikube (nework_delete)$ docker network create --driver=bridge --subnet=192.168.49.0/24 -o com.docker.network.bridge.enable_ip_masquerade=true --label=created_by.minikube.sigs.k8s.io=true p3
Error response from daemon: Pool overlaps with other one on this address space
med@xmac:~/workspace/minikube (nework_delete)$ echo $?
1
```